Makefile

```diff
@@ -72,51 +70,48 @@ tools: $(CROSSLINK) $(GOLANGCI_LINT) $(MISSPELL) $(GOCOVMERGE) $(STRINGER) $(POR
 
 # Build
 
-.PHONY: examples generate build
-examples:
-	@set -e; for dir in $(EXAMPLES); do \
-	  echo "$(GO) build $${dir}/..."; \
-	  (cd "$${dir}" && \
-	   $(GO) build .); \
-	done
-
-generate: $(STRINGER) $(PORTO)
-	set -e; for dir in $(ALL_GO_MOD_DIRS); do \
-	  echo "$(GO) generate $${dir}/..."; \
-	  (cd "$${dir}" && \
-	    PATH="$(TOOLS):$${PATH}" $(GO) generate ./... && \
-		$(PORTO) -w .); \
-	done
-
-build: generate
-	# Build all package code including testing code.
-	set -e; for dir in $(ALL_GO_MOD_DIRS); do \
-	  echo "$(GO) build $${dir}/..."; \
-	  (cd "$${dir}" && \
-	    $(GO) build ./... && \
-		$(GO) list ./... \
-		  | grep -v third_party \
-		  | xargs $(GO) test -vet=off -run xxxxxMatchNothingxxxxx >/dev/null); \
-	done
+.PHONY: generate build
+
+generate: $(OTEL_GO_MOD_DIRS:%=generate/%)
+generate/%: DIR=$*
+generate/%: | $(STRINGER) $(PORTO)
+	@echo "$(GO) generate $(DIR)/..." \
+		&& cd $(DIR) \
+		&& PATH="$(TOOLS):$${PATH}" $(GO) generate ./... && $(PORTO) -w .
+
+build: generate $(OTEL_GO_MOD_DIRS:%=build/%) $(OTEL_GO_MOD_DIRS:%=build-tests/%)
+build/%: DIR=$*
+build/%:
+	@echo "$(GO) build $(DIR)/..." \
+		&& cd $(DIR) \
+		&& $(GO) build ./...
+
+build-tests/%: DIR=$*
+build-tests/%:
+	@echo "$(GO) build tests $(DIR)/..." \
+		&& cd $(DIR) \
+		&& $(GO) list ./... \
+		| grep -v third_party \
+		| xargs $(GO) test -vet=off -run xxxxxMatchNothingxxxxx >/dev/null
 
 # Tests
 
 TEST_TARGETS := test-default test-bench test-short test-verbose test-race
 .PHONY: $(TEST_TARGETS) test
-test-default: ARGS=-v -race
+test-default: ARGS=-race
 test-bench:   ARGS=-run=xxxxxMatchNothingxxxxx -test.benchtime=1ms -bench=.
 test-short:   ARGS=-short
 test-verbose: ARGS=-v
 test-race:    ARGS=-race
```

Comment:
If the default is -race, do we need a `test-race`?  Also shouldn't verbose also have the -race?

Reply:
I can combine the `test-race` and `test-default` targets. I think keeping both doesn't hurt, especially if new devs are not familiar with what default is.